be1-go/hub/standard_hub/message_handling.go

```diff
@@ -468,11 +469,11 @@ func (h *Hub) handleGreetServer(socket socket.Socket, byteMessage []byte) error
 
 // getMissingIds compares two maps of channel Ids associated to slices of message Ids to
 // determine the missing Ids from the storedIds map with respect to the receivedIds map
-func getMissingIds(receivedIds map[string][]string, storedIds map[string][]string, blacklist []string) map[string][]string {
+func getMissingIds(receivedIds map[string][]string, storedIds map[string][]string, blacklist *hub_state.ThreadSafeSlice[string]) map[string][]string {
```

Comment:
Do you really need to give a pointer for blacklist?

Reply:
If I pass blacklist as a value, a copy of the Mutex will be made along with the struct, leading to unexpected behaviours

Reply:
you are right mybad